run_lint.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+START=f142077094608ceb37f7605d2c4f53243be4fb22
+END=HEAD
+
+DIFF=`git diff --name-only $START $END`
+COMMITS=`git log --format=%H $START..$END`
+
+GREP_FILTER="^("
+for d in $COMMITS; do
+  GREP_FILTER+=$d
+  GREP_FILTER+="|"
+done
+
+GREP_FILTER=${GREP_FILTER::-1}
+GREP_FILTER+=")"
```

Comment:
I like the idea in principle, but because the porcelain format has a load of additional information it seems like you first have to grep all the lines that look like lines then grep away the ones we don't want. I've done this but I'm not so certain it is better so I've not squashed it yet. What do you think?